cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -314,6 +336,13 @@ ClusterOperationMetadata createCluster(String name, String imageVersion, Map<Str
                                          boolean privateInstance)
     throws RetryableProvisionException, InterruptedException, IOException {
 
+    if (network == null) {
+      // This shouldn't happen as the fromConf method should already check.
+      // This is to guard against programmatic bug that this client was created without network information and
+      // yet being used to create cluster.
+      throw new IllegalArgumentException("Missing network information");
```

Comment:
Should we add it to @throws? Also I'd reformulate "This shouldn't happen" as it may happen, it's just a client that can't create clusters.

Reply:
This is a package private class which is only used privately by the provisioner. This is just a simple check to fail early in test in case someone wrote a buggy code within this package.

Reply:
This is more like an assert that shouldn't be declared as part of the API contract.